mlir/tools/rocmlir-gen/rocmlir-gen.cpp

```diff
@@ -1884,6 +1914,68 @@ static func::FuncOp createGpuGemmKernel(ModuleOp module,
   return func;
 }
 
+static void getAttentionTypes(ArrayRef<Type> elemTypes,
+                              SmallVectorImpl<Type> &result) {
+
+  // Note: the vector of `elemTypes` contains the same types.
+  // This behaviour enforced by detectMissingArguments()
+
+  const size_t typeIndex{0};
+  SmallVector<int64_t> dims{sequenceLength, headDims};
+  SmallVector<int64_t> scaleDims{sequenceLength, headDims};
+
+  MemRefType qType = MemRefType::get(dims, elemTypes[typeIndex]),
+             kType = MemRefType::get(dims, elemTypes[typeIndex]),
+             vType = MemRefType::get(dims, elemTypes[typeIndex]),
+             sType = MemRefType::get(scaleDims, elemTypes[typeIndex]),
+             outType = MemRefType::get(dims, elemTypes[typeIndex]);
+
+  result.push_back(qType);
+  result.push_back(kType);
+  result.push_back(vType);
+  result.push_back(sType);
+  result.push_back(outType);
+}
+
+static func::FuncOp createGpuAttentionKernel(ModuleOp module,
+                                             const GenParams &params) {
+  MLIRContext *ctx = module.getContext();
+  Location loc = module->getLoc();
+  OpBuilder builder(ctx);
+
+  // Set mhal.arch on module to make compilation pipeline work
+  StringAttr archAttr = builder.getStringAttr(params.arch);
+  if (!module->hasAttr("mhal.arch"))
+    module->setAttr("mhal.arch", archAttr);
+
+  SmallVector<Type, 5> argTypes;
+  getAttentionTypes(params.types, argTypes);
+
+  SmallVector<NamedAttribute, 2> funcAttrs = {
+      builder.getNamedAttr("kernel", builder.getUnitAttr()),
+      builder.getNamedAttr("mhal.arch", archAttr)};
+
+  constexpr StringLiteral kernelName("rocm_attention");
```

Comment:
not : lets call it rock_attention as rock is our project name and also being consistent with others.

Reply:
sorry, it was typo. I wanted to call it `rock`